cabot/plugin_test_utils.py

```diff
@@ -0,0 +1,230 @@
+            super(MyCoolPluginTestCase, self).setUp()
+
+            # add the email alert to the test service
+            self.email_alert = EmailAlert.objects.get(title=models.EmailAlert.name)
+            self.service.alerts.add(self.email_alert)
+            self.service.save()
+
+    This class provides two helper functions to help with testing:
+        transition_service(old_status, new_status)
+    and
+        run_checks([(check, passed, acked), ...], from_service_status)
+
+    If your alert type builds a message based on the state of a service's checks (which most plugins do), you should
+    probably use run_checks() to simulate failing checks and verify you build your message as expected.
+    Otherwise, you can use transition_service() for a simple Service status-based test.
+
+    For example, to run the Jenkins, HTTP, and TCP checks, with the states failing, acked,
+    then updating the service (coming from the 'PASSING' status):
+
+        run_checks([
+            (self.jenkins_check, False, False),
+            (self.http_check, False, True),
+            (self.tcp_check, True, True)],
+            Service.PASSING_STATUS)
+
+    Or you can just test with the service status, if you don't care about check states:
+
+        transition_service(Service.PASSING_STATUS, Service.ERROR_STATUS)
+
+    You should probably at least test the following situations:
+    * passing -> warning
+    * passing -> error
+    * error -> passing
+    * error -> acked
+    """
+    def setUp(self):
+        update_alert_plugins()
+
+        self.user = User.objects.create(username='testuser', password='testuserpassword', email='test@test.test')
+        self.user.user_permissions.add(
+            Permission.objects.get(codename='add_service'),
+            Permission.objects.get(codename='add_httpstatuscheck'),
+            Permission.objects.get(codename='add_jenkinsstatuscheck'),
+            Permission.objects.get(codename='add_tcpstatuscheck'),
+        )
+        self.user.save()
+
+        self.jenkins_check = JenkinsStatusCheck.objects.create(
+            id=10101,
+            name='Jenkins Check',
+            created_by=self.user,
+            importance=Service.ERROR_STATUS,
+            max_queued_build_time=10,
+            max_build_failures=5
+        )
+
+        self.http_check = HttpStatusCheck.objects.create(
+            id=10102,
+            name='Http Check',
+            created_by=self.user,
+            importance=Service.CRITICAL_STATUS,
+            endpoint='https://google.com',
+            timeout=10,
+            status_code='200',
+            text_match=None,
+        )
+        self.tcp_check = TCPStatusCheck.objects.create(
+            id=10103,
+            name='TCP Check',
+            created_by=self.user,
+            importance=Service.ERROR_STATUS,
+            address='github.com',
+            port=80,
+            timeout=6,
+        )
```

Comment:
add elasticsearch also? I thought some of the email/hipchat tests at least had checks for images, which are only generated for es checks (and if they don't they should haha)

Reply:
Yeah, I definitely should've added something to test images. Will do (then update the plugin PRs accordingly).